app/controllers/api/imports.php

```diff
@@ -0,0 +1,403 @@
+    ->label('sdk.auth', [APP_AUTH_TYPE_KEY])
+    ->label('sdk.namespace', 'imports')
+    ->label('sdk.method', 'retry')
+    ->label('sdk.description', '/docs/references/imports/retry-import.md')
+    ->label('sdk.response.code', Response::STATUS_CODE_OK)
+    ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
+    ->label('sdk.response.model', Response::MODEL_IMPORT)
+    ->param('importId', '', new UID(), 'Import unique ID.')
+    ->inject('response')
+    ->inject('dbForProject')
+    ->inject('project')
+    ->inject('user')
+    ->inject('events')
+    ->action(function (string $importId, Response $response, Database $dbForProject, Document $project, Document $user, Event $eventInstance) {
+        $import = $dbForProject->getDocument('imports', $importId);
+
+        if ($import->isEmpty()) {
+            throw new Exception(Exception::IMPORT_NOT_FOUND);
+        }
+
+        // if ($import->getAttribute('status') !== 'failed') {
+        //     throw new Exception(Exception::IMPORT_IN_PROGRESS, 'Import not failed');
+        // }
+
+        $import
+            ->setAttribute('status', 'pending')
+            ->setAttribute('dateUpdated', \time());
+
+        // Trigger Import
+        $event = new Import();
+        $event
+            ->setImport($import)
+            ->setProject($project)
+            ->setUser($user)
+            ->trigger();
+
+        $response->noContent();
+    });
+
+App::delete('/v1/imports/:importId')
+    ->groups(['api', 'imports'])
+    ->desc('Delete Import')
+    ->label('scope', 'imports.write')
+    ->label('event', 'imports.[importId].delete')
+    ->label('audits.event', 'importId.delete')
+    ->label('audits.resource', 'imports/{request.importId}')
+    ->label('sdk.auth', [APP_AUTH_TYPE_KEY])
+    ->label('sdk.namespace', 'imports')
+    ->label('sdk.method', 'delete')
+    ->label('sdk.description', '/docs/references/functions/delete-import.md')
+    ->label('sdk.response.code', Response::STATUS_CODE_NOCONTENT)
+    ->label('sdk.response.model', Response::MODEL_NONE)
+    ->param('importId', '', new UID(), 'Import ID.')
+    ->inject('response')
+    ->inject('dbForProject')
+    ->inject('deletes')
+    ->inject('events')
+    ->action(function (string $importId, Response $response, Database $dbForProject, Delete $deletes, Event $events) {
+
+        $import = $dbForProject->getDocument('imports', $importId);
+
+        if ($import->isEmpty()) {
+            throw new Exception(Exception::IMPORT_NOT_FOUND, 'Import not found', 404);
+        }
+
+        if (!$dbForProject->deleteDocument('imports', $import->getId())) {
+            throw new Exception(Exception::GENERAL_SERVER_ERROR, 'Failed to remove import from DB', 500);
+        }
+
+        $deletes
+            ->setType(DELETE_TYPE_DOCUMENT)
+            ->setDocument($import);
+
+        $events->setParam('importId', $import->getId());
+
+        $response->noContent();
+    });
+
+App::post('/v1/imports/appwrite')
```

Comment:
Let's add all CREATE operation at the start of the controller.

Reply:
I'm not sure I understand what you mean by a CREATE operation?